dragonfly/block/lantern.go

```diff
@@ -0,0 +1,90 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/block/model"
+	"github.com/df-mc/dragonfly/dragonfly/item"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/go-gl/mathgl/mgl64"
+)
+
+// Lantern is a light emitting block.
+type Lantern struct {
+	noNBT
+	transparent
+
+	// Hanging determines if a lantern is hanging off a block.
+	Hanging bool
+	// Soul determines whether it is a normal lantern or soul lantern.
+	Soul bool
```

Comment:
I feel like it might be a better idea to have a separate `torch type` (like with corals) in case extra types of torches/lanterns are added. You could put the light level in there as well so that we can re-use it for torches.